row: add debugging for TestExternalRowData

michae2 opened the pull request:
When modifying keys in the BatchRequest for external row data, we sometimes see that the key has already been modified. We suspect that multiple keys in the BatchRequest share an underlying array. Change the crdb-test-build-only assertion to a panic to get a stacktrace during this rare test failure.

Informs: #128070

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/131048)
<!-- Reviewable:end -->

mgartner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/131048#-:-O7Awig4-q-bnwNxwTzL:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mw5h)



<!-- Sent from Reviewable.io -->

mw5h commented:
:lgtm:

michae2 committed:
row: add debugging for TestExternalRowData

When modifying keys in the BatchRequest for external row data, we
sometimes see that the key has already been modified. We suspect that
multiple keys in the BatchRequest share an underlying array. Change the
crdb-test-build-only assertion to a panic to get a stacktrace during
this rare test failure.

Informs: #128070

Release note: None

michae2 commented:
Thanks!

bors r=mgartner,mw5h

msbutler commented:
@michae2 looks like @fqazi [new test case ](https://github.com/cockroachdb/cockroach/pull/131321#pullrequestreview-2337433433)hit this stack trace here https://teamcity.cockroachdb.com/viewLog.html?buildId=17076133&buildTypeId=Cockroach_BazelExtendedCi

mgartner commented:
We also hit it here: https://github.com/cockroachdb/cockroach/issues/131719
